test/framework/prometheus.go

```diff
@@ -405,6 +405,10 @@ func (f *Framework) WaitForPrometheusReady(ctx context.Context, p *monitoringv1.
 			if cond.Type == monitoringv1.PrometheusReconciled {
 				reconciled = &cond
 			}
+			if cond.ObservedGeneration != current.Generation {
+				pollErr = errors.Errorf("prometheus generation is not equal to conditions observed generation")
```

Comment:
let's report the exact problem.
```suggestion
				pollErr = errors.Errorf("observed generation %d for condition %s isn't equal to the state generation %d", cond.ObservedGeneration, cond.Type, current.Generaiton)
```